airflow/providers/microsoft/azure/operators/batch.py

```diff
@@ -293,16 +293,17 @@ def execute(self, context: Context) -> None:
         # Add task to job
         self.hook.add_single_task_to_job(job_id=self.batch_job_id, task=task)
         # Wait for tasks to complete
-        fail_tasks = self.hook.wait_for_job_tasks_to_complete(job_id=self.batch_job_id, timeout=self.timeout)
+        failed_tasks = self.hook.wait_for_job_tasks_to_complete(job_id=self.batch_job_id,
+                                                                timeout=self.timeout)
         # Clean up
         if self.should_delete_job:
             # delete job first
             self.clean_up(job_id=self.batch_job_id)
         if self.should_delete_pool:
             self.clean_up(self.batch_pool_id)
         # raise exception if any task fail
```

Comment:
It's odd that you're waiting on several tasks even though only one was submitted, but then again, I'm not very familiar with this Azure API.

Either way, this comment should be updated to reflect the new behaviour of only failing on a specific task.

Reply:
Good point. I wrote an improvement locally, let me test it and then push it.

Reply:
I've refactored it @o-nikolas feel free to take a look.